No man page for ed, the standard text editor

## Issue description

There seems to be no man page for ed, the standard text editor.

### Steps to reproduce

```
$ nix-shell --pure -p ed -p man -p less \
    -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/master.tar.gz
$ man ed
No manual entry for ed
```

## Technical details

```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.58, NixOS, 17.09.1880.ac2bb5684c (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.15`
 - channels(root): `"nixos-17.09.1880.ac2bb5684c"`
 - channels(kristoffer): `"nixpkgs-17.09pre112229.e619ace733"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
```

`pkgs/applications/editors/ed/default.nix` contains:
```
     /* FIXME: Tests currently fail on Darwin:

       building test scripts for ed-1.5...
       testing ed-1.5...
       *** Output e1.o of script e1.ed is incorrect ***
       *** Output r3.o of script r3.ed is incorrect ***
       make: *** [check] Error 127

    */
```
I suspect the next code results in double application of the nix store directory:

```
  installFlags = [ "DESTDIR=$(out)" ];

  configureFlags = [
    "--exec-prefix=${stdenv.cc.prefix}"
    "CC=${stdenv.cc.prefix}cc"
  ];
```

I 'll investigate further.

This might be related to a comment in `pkgs/applications/editors/ed/default.nix`:
```
     /* FIXME: Tests currently fail on Darwin:

       building test scripts for ed-1.5...
       testing ed-1.5...
       *** Output e1.o of script e1.ed is incorrect ***
       *** Output r3.o of script r3.ed is incorrect ***
       make: *** [check] Error 127

    */
```
The target directory is set with:

```
  installFlags = [ "DESTDIR=$(out)" ];

  configureFlags = [
    "--exec-prefix=${stdenv.cc.prefix}"
    "CC=${stdenv.cc.prefix}cc"
  ];
```
There's another configure flag, `--prefix`, that defaults to `/usr/local`. I think nix automatically replaces that by `$(out)`. Resulting in `$(out)/$(out)`.
`--exec-prefix` would default to '$(prefix)`, but since it was set, the bin files ended up in the right place.

Not setting any installFlags results in a correct tree. It basically lets nix figure it out.
```
  # installFlags = [ "DESTDIR=$(out)" ];

  configureFlags = [
    "--exec-prefix=${stdenv.cc.prefix}"
    "CC=${stdenv.cc.prefix}cc"
  ];
```
I noticed @Ericson2314 very recently replaced `stdenv.cc.prefix` by `stdenv.cc.targetPrefix'.

Todo:
- [ ] Confirm it's ok to not set the `DESTDIR`
- [ ] Check if the `FIXME` comment in the source about Darwin is solved.

--- a/pkgs/applications/editors/ed/default.nix
+++ b/pkgs/applications/editors/ed/default.nix
@@ -13,16 +13,7 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ lzip ];
 
-  /* FIXME: Tests currently fail on Darwin:
-
-       building test scripts for ed-1.5...
-       testing ed-1.5...
-       *** Output e1.o of script e1.ed is incorrect ***
-       *** Output r3.o of script r3.ed is incorrect ***
-       make: *** [check] Error 127
-
-    */
-  doCheck = !(hostPlatform.isDarwin || hostPlatform != buildPlatform);
+  doCheck = hostPlatform == buildPlatform;
 
   meta = {
     description = "An implementation of the standard Unix editor";

--- a/pkgs/applications/editors/ed/default.nix
+++ b/pkgs/applications/editors/ed/default.nix
@@ -11,8 +11,6 @@ stdenv.mkDerivation rec {
     sha256 = "1nqhk3n1s1p77g2bjnj55acicsrlyb2yasqxqwpx0w0djfx64ygm";
   };
 
-  unpackCmd = "tar --lzip -xf";
-
   nativeBuildInputs = [ lzip ];
 
   /* FIXME: Tests currently fail on Darwin:

--- a/pkgs/applications/editors/ed/default.nix
+++ b/pkgs/applications/editors/ed/default.nix
@@ -26,13 +26,6 @@ stdenv.mkDerivation rec {
     */
   doCheck = !(hostPlatform.isDarwin || hostPlatform != buildPlatform);
 
-  installFlags = [ "DESTDIR=$(out)" ];
-
-  configureFlags = [
-    "--exec-prefix=${stdenv.cc.targetPrefix}"
-    "CC=${stdenv.cc.targetPrefix}cc"
-  ];
-
   meta = {
     description = "An implementation of the standard Unix editor";
 

--- a/pkgs/tools/compression/lzip/default.nix
+++ b/pkgs/tools/compression/lzip/default.nix
@@ -13,7 +13,10 @@ stdenv.mkDerivation rec {
 
   configureFlags = "CPPFLAGS=-DNDEBUG CFLAGS=-O3 CXXFLAGS=-O3";
 
+  setupHook = ./lzip-setup-hook.sh;
+
   doCheck = true;
+  enableParallelBuilding = true;
 
   meta = {
     homepage = http://www.nongnu.org/lzip/lzip.html;

--- a/pkgs/tools/compression/lzip/lzip-setup-hook.sh
+++ b/pkgs/tools/compression/lzip/lzip-setup-hook.sh
@@ -1,0 +1,5 @@
+lzipUnpackCmdHook() {
+    [[ "$1" = *.tar.lz ]] && tar --lzip -xf "$1"
+}
+
+unpackCmdHooks+=(lzipUnpackCmdHook)

